mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -652,23 +652,25 @@ def _get_attributes_filtering_clauses(parsed):
         key_type = sql_statement.get('type')
         key_name = sql_statement.get('key')
         value = sql_statement.get('value')
-        comparator = sql_statement.get('comparator')
+        comparator = sql_statement.get('comparator').upper()
         if SearchUtils.is_attribute(key_type, comparator):
-            # validity of the comparator is checked in SearchUtils.parse_search_filter()
-            op = SearchUtils.filter_ops.get(comparator)
-            if op:
-                # key_name is guaranteed to be a valid searchable attribute of entities.RunInfo
-                # by the call to parse_search_filter
-                attribute_name = SqlRun.get_attribute_name(key_name)
-                clauses.append(op(getattr(SqlRun, attribute_name), value))
+            # key_name is guaranteed to be a valid searchable attribute of entities.RunInfo
+            # by the call to parse_search_filter
+            attribute = getattr(SqlRun, SqlRun.get_attribute_name(key_name))
+            if comparator in SearchUtils.CASE_INSENSITIVE_STRING_COMPARISON_OPERATORS:
```

Comment:
Can we add documentation on LIKE and ILIKE, for [this page](https://mlflow.org/docs/latest/search-syntax.html#id9)? We should also either clarify that LIKE may also be case-insensitive depending on the circumstance (e.g., sqlite is case-insensitive) or somehow make it always case-sensitive (seems preferable if possible).

Reply:
Sure, I'll add documentation when I update my PR after a new release of sqlparse (the pr is kind of blocked waiting for that)

Reply:
The author of that repo said there should be a new release by the end of the month. Since this is not super urgent, I guess we can wait for that.